_includes/tutorials/schedule-ktable-ttl-ii/kstreams/code/src/main/java/io/confluent/developer/serdes/JSONSerde.java

```diff
@@ -0,0 +1,64 @@
+package io.confluent.developer.serdes;
+
+import java.io.IOException;
+import java.util.Map;
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serializer;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class JSONSerde<T> implements Serializer<T>, Deserializer<T>, Serde<T>  {
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  @Override
+  public Serializer<T> serializer() {
+    return this;
+  }
+
+  @Override
+  public Deserializer<T> deserializer() {
+    return this;
+  }
+
+  @Override
+  public T deserialize(String topic, byte[] data) {
+    if(data == null) return null;
+    T result;
+    try {
+      result = (T) OBJECT_MAPPER.readValue(data, JSONSerdeCompatible.class);
+      System.out.println("Serialized = "+result);
+    } catch (JsonParseException e) {
+      throw new SerializationException(e);
+    } catch (JsonMappingException e) {
+      throw new SerializationException(e);
+    } catch (IOException e) {
+      throw new SerializationException(e);
+    }
```

Comment:
nit: Since all the catch blocks do the same action consider simplifying it to one (both of the JSON exceptions extend IOException)

```java
} catch (IOException e) {
      throw new SerializationException(e); 
}
```